android/src/main/java/com/bitmovin/player/reactnative/offline/OfflineManager.kt

```diff
@@ -0,0 +1,208 @@
+import com.bitmovin.player.api.offline.OfflineContentManager
+import com.bitmovin.player.api.offline.OfflineContentManagerListener
+import com.bitmovin.player.api.offline.options.OfflineContentOptions
+import com.bitmovin.player.api.offline.options.OfflineOptionEntry
+import com.bitmovin.player.api.offline.options.OfflineOptionEntryAction
+import com.bitmovin.player.api.offline.options.OfflineOptionEntryState
+import com.bitmovin.player.api.source.SourceConfig
+import com.bitmovin.player.reactnative.NativeId
+import com.bitmovin.player.reactnative.converter.JsonConverter
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.modules.core.DeviceEventManagerModule
+
+class OfflineManager(
+    private val nativeId: NativeId,
+    private val context: ReactApplicationContext,
+    private val offlineId: String,
+    source: SourceConfig,
+    location: String
+) :
+    OfflineContentManagerListener {
+
+    val contentManager: OfflineContentManager
+    var contentOptions: OfflineContentOptions? = null
+        private set
+
+    init {
+        contentManager = OfflineContentManager.getOfflineContentManager(
+            source, location, offlineId, this, context
+        )
+    }
+
+    fun getOptions() {
+        contentManager.getOptions()
+    }
+
+    /**
+     * Process the `OfflineDownloadRequest`.
+     * The `OfflineContentOptions` are stored in memory in this class because they can not be constructed to pass to the process call.
+     * We're copying the iOS interface for the `minimumBitrate` so that react-native will have a consistent interface.
+     *
+     */
+    fun process(request: OfflineDownloadRequest) {
+        if (contentOptions != null) {
+            contentOptions!!.videoOptions
+                .firstOrNull { option -> option.bitrate >= request.minimumBitrate }
+                ?.let { option -> option.action = OfflineOptionEntryAction.Download }
+            changeToDownloadAction(request.audioOptionIds, contentOptions!!.audioOptions)
+            changeToDownloadAction(request.textOptionIds, contentOptions!!.textOptions)
+
+            contentManager.process(contentOptions!!)
+        }
+    }
+
+    private fun changeToDownloadAction(
+        ids: List<String?>?,
+        potentialOptions: List<OfflineOptionEntry>
+    ) {
+        if (ids.isNullOrEmpty()) {
+            return
+        }
+
+        ids.forEach { idToDownload ->
+            potentialOptions.forEach { option ->
+                if (idToDownload == option.id && option.action !== OfflineOptionEntryAction.Download) {
+                    option.action = OfflineOptionEntryAction.Download
+                }
+            }
+        }
+    }
+
+    fun resume() {
+        contentManager.resume()
+    }
+
+    fun suspend() {
+        contentManager.suspend()
+    }
+
+    fun cancelDownload() {
+        suspend()
+        deleteAll()
```

Comment:
`deleteAll` will delete everything and probably is not what what somebody would expect when canceling downloads 🤔 

I think correctly, you will need to get the current `OfflineContentOptions` check for what is `Downloading`,`Suspended` or `Failed` and set the `action` of these options to `Delete`.

Reply:
After a talk to @rolandkakonyi I think this is fine.

Reply:
@rolandkakonyi @matamegger Now that i'm back to this, are we still ok with deleting the offline content associated with the specific offline id that this `OfflineManager` is constructed with?  It's been a little while, but I believe I made this decision because that's what iOS seemed to do under the hood, so it felt like deleting everything for that id was a good spot to leave the user in.

Reply:
@joornby-angel I am fine with it if we need to make it behave the same for both.
If you want to pause a download, you suspend it. If you want to terminate it, you cancel it. On iOS canceling (=termination) means you delete any partially downloaded data as you can't continue after canceled. One must start a new download after cancelation.

@matamegger do you have any additional thoughts on this?

Reply:
Well summarized @rolandkakonyi. It should behave the same. So this is correct! 🙂